### PR TITLE
feat: updates multihash table

### PR DESCRIPTION
**Pull request by hugomrdias**
- adds script to auto update
- reduces bundle size by having only one explicit table ( 3kb gziped diff )
- remove defaultLengths seems to not be used and multicodec tables doesn't have it

closes https://github.com/multiformats/js-multihash/issues/73

BREAKING CHANGE: defaultLengths removed, skein is now lowercase

### CODE REVIEW

**Comment by rvagg** [comment]
also breaks by changing case of some of the names, like `Skein` -> `skein`

but very good change though! I wondered why this was manually replicated and I guess that's just historical.

File: src/constants.js [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
-  'sha3-384':   0x15,
-  'sha3-512':   0x14,
-  'shake-128':  0x18,
-  'shake-256':  0x19,
-  'keccak-224': 0x1A,
-  'keccak-256': 0x1B,
-  'keccak-384': 0x1C,
-  'keccak-512': 0x1D,
+const names = Object.freeze({
+  'identity': 0x00,
```
[the remark this answers is not in the document]
Comment (hugomrdias) [reply src/constants.js:5]: i don't think it matter much and this is how it comes from the multicodec source table i would rather keep it simple and lose a couple of bytes.

[the remark this answers is not in the document]
Comment (vmx) [reply src/constants.js:5]: A benefit of parsing the number from the table would be that we can make sure they are actually well-formed. But I don't have a strong opinion on it, feel free to keep it as hex.

File: update-table.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+'use strict'
```
Comment (jacobheun) [inline update-table.js:1]: Let's document the existence and usage of this in the readme

**Comment by vmx** [comment]
@hugomrdias could you please force-push this commit again with an updated commit message that also makes the "Skein" casing change as breaking change (this way it should show up nicely in the changelog). It could be done while merging, but i also forgot doing those things quite often :)

**Comment by hugomrdias** [comment]
@jacobheun FYI changed the maintainer to me

**Comment by achingbrain** [comment]
This contains a breaking change, it should have gone out as `v0.5.0`.

`ipld-git` is currently broken as it depends on `~0.4.14` of this module and accesses the `defaultLengths` property from `/src/constants`.

Please can you re-release `0.4.19` as `0.4.21` then release `0.4.20` as `0.5.0`.

**Pull request merged**